cmake/Modules/CUDAQConfig.cmake

```diff
@@ -78,5 +78,45 @@ set_target_properties(cudaq::cudaq-builder PROPERTIES
   IMPORTED_LOCATION "${CUDAQ_LIBRARY_DIR}/libcudaq-builder${CMAKE_SHARED_LIBRARY_SUFFIX}"
   IMPORTED_SONAME "libcudaq-builder${CMAKE_SHARED_LIBRARY_SUFFIX}")
 
-set(CUDAQ_TARGET "nvidia" CACHE STRING "The CUDA Quantum target to compile for and execute on. Defaults to `nvidia`")
+# Check for the presence of NVIDIA GPUs, if none
+# are found set the default target to qpp-cpu
+set(__tmp_cudaq_target "qpp-cpu")
+include(CheckLanguage)
+check_language(CUDA)
+if(CMAKE_CUDA_COMPILER)
+  enable_language(CUDA)
+  message(STATUS "CUDA compiler found: ${CMAKE_CUDA_COMPILER}")
+  set(CUDA_TEST_SOURCE "
+    #include <cuda_runtime.h>
+    #include <stdio.h>
+    int main() {
+      int deviceCount;
+      cudaError_t error = cudaGetDeviceCount(&deviceCount);
```

Comment:
Just curious ... why not use `nvidia-smi -L | wc -l` instead of this program?

Reply:
updated